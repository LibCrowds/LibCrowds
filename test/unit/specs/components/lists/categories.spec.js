import BootstrapVue from 'bootstrap-vue'

import testLocalConfig from '~/test/test.local.config'
import { mount, createLocalVue } from 'vue-test-utils'
import CategoriesList from '~/components/lists/Categories'

describe('Categories list', () => {
  let localVue = null
  let categories = null
  let wrapper = null
  let changeCategorySpy = null

  beforeEach(() => {
    localVue = createLocalVue()
    localVue.use(BootstrapVue)
    categories = [
      {
        id: 1,
        name: 'london'
      },
      {
        id: 2,
        name: 'plymouth'
      }
    ]
    changeCategorySpy = spyOn(CategoriesList.methods, 'changeCategory')
    wrapper = mount(CategoriesList, {
      localVue,
      propsData: {
        categories: categories,
        header: 'Locations'
      }
    })
  })

  it('renders correctly', () => {
    const renderer = require('vue-server-renderer').createRenderer()
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })

  it('changes to the first category when mounted', () => {
    expect(changeCategorySpy).toHaveBeenCalledWith(categories[0])
  })

  it('emits change event and sets activeCategory on category change', () => {
    wrapper.vm.changeCategory(categories[1])
    expect(wrapper.vm.activeCategory).toBe(categories[1])
    expect(wrapper.emitted()).toBe({
      'change': categories[1]
    })
  })
})
