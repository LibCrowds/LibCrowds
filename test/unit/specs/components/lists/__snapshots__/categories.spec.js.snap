// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories list renders correctly 1`] = `

<div id="categories-list"
     data-server-rendered="true"
     class="card"
>
  <div class="card-header">
    <div>
      Locations
    </div>
  </div>
  <div class="list-group">
    <div class="list-group-item list-group-item-action">
      london
    </div>
    <div class="list-group-item list-group-item-action">
      plymouth
    </div>
  </div>
</div>

`;
